### ISSUE TITLE
Additional control over the "rate"

### ISSUE THREAD

**Issue by zyddnys** [tqdm/tqdm#295]
I think there can be a rate_scale parameter so the rate calculated can be multiplied by the rate_scale parameter(rate*=rate_scale), therefore user can have better control over "rate" displayed.

For example:
BLOCK_SIZE = 1024*1024*256
# before
for chunk in tqdm( split_file( 'largefile.bin', BLOCK_SIZE ), unit = 'chunk' ) :
    # do somthing..
# after
for chunk in tqdm( split_file( 'largefile.bin', BLOCK_SIZE ), unit = 'bytes', rate_scale = BLOCK_SIZE, unit_scale = True ) :
    # do somthing..

**Comment by lrq3000** [comment]
I don't understand what is the purpose, can you provide more details about
the target output you expect? Are you really talking about the internal
rate (which is used for the smoothing of miniters) or the displayed rate
(which is the number of iterations per second)?

2016-10-25 14:00 GMT+02:00 zyddnys <notifications@github.com>:

**Comment by zyddnys** [comment]
Sorry for my poor expression, I mean the displayed rate
(which is the number of iterations per second)

**Comment by lrq3000** [comment]
Don't worry it's not your expression, it's just that i need kore details :)

And unit scale = True is not enough? Can you show an example code with
output you currently get vs what you would like?

**Comment by zyddnys** [comment]
Sorry for being late to reply.
I haven't fully gone through the codes yet, but the current changes I have made haven't cause any problem in my programs since I only use a portion of tqdm.
I don't know if the changes I made could cause problem in other functions of tqdm, but anyway here are some changes I have made :
# Changes
* In the constructor function of tqdm and, new paramater ``rate_scale`` was added and stored
```python
def __init__(self, iterable=None, desc=None, total=None, leave=True,
                  file=sys.stderr, ncols=None, mininterval=0.1,
                  maxinterval=10.0, miniters=None, ascii=None, disable=False,
                  unit='it', unit_scale=False, dynamic_ncols=False,
                  smoothing=0.3, bar_format=None, initial=0, position=None,
                  gui=False, rate_scale=1, **kwargs):
    # Store the arguments
    # ...
    self.bar_format = bar_format
    self.rate_scale = rate_scale
    # ...
```
* In the ``format_meter`` function, ``rate_scale`` was added, and later was used to scale the rate been displayed
```python
     @staticmethod
     def format_meter(n, total, elapsed, ncols=None, prefix='',
                      ascii=False, unit='it', unit_scale=False, rate=None,
                      bar_format=None, rate_scale=1):
         # ...
         # if unspecified, attempt to use rate = average speed
         # (we allow manual override since predicting time is an arcane art)
         if rate is None and elapsed :
             rate = n / elapsed
         if not rate is None :
             rate *= rate_scale
         # ...
```
* And of course all function calls to ``format_meter``, ``self.rate_scale`` was added in the parameters
```python
self.format_meter(self.n, total, 0, 
                  (dynamic_ncols(file) if dynamic_ncols else ncols),
                  self.desc, ascii, unit, unit_scale, None, bar_format,
                  self.rate_scale)
```
# Outputs
* Before
```python
BLOCK_SIZE = 1024 * 1024 * 256
for chunk in tqdm( split_file( 'largefile.bin', BLOCK_SIZE ), unit = 'chunk' ) :
	# do something..
```
> 76%|████████████████████████████         | 7568/10000 [00:33<00:10, 2.00chunk/s]

* After
```python
BLOCK_SIZE = 1024 * 1024 * 256
for chunk in tqdm( split_file( 'largefile.bin', BLOCK_SIZE ), unit = 'Bytes', rate_scale = BLOCK_SIZE, unit_scale = True ) :
	# do something..
```
> 76%|████████████████████████████         | 7568/10000 [00:33<00:10, 512.00MBytes/s]
#### Rate displayed changed from ``2.00chunk/s`` to ``512.00MBytes/s``

**Comment by casperdcl** [comment]
What about this? Similar to how the [urllib example](https://github.com/tqdm/tqdm/blob/master/examples/tqdm_wget.py) works.
``` python
BLOCK_SIZE = 1024 * 1024 * 256
chunks = split_file("largefile.bin", BLOCK_SIZE)
with tqdm(total=len(chunks), unit="Bytes", unit_scale=True) as t:
  for chunk in chunks:
    # do something..
    t.update(BLOCK_SIZE)
```

**Comment by casperdcl** [comment]
There's also this for further reading: https://github.com/tqdm/tqdm/wiki/How-to-make-a-great-Progress-Bar

Not sure whether we were going to reintroduce that section into the README @lrq3000 ...

**Comment by zyddnys** [comment]
Great piece of advice, getting things done in one line is great but it's not always the case

**Comment by lrq3000** [comment]
@zyddnys Casper gave you the link and an example about how to manage these cases, you should use `tqdm.update()` and not the iterable based tqdm.

However, thank you for your suggestion and code, that is very interesting, because iterable based tqdm cannot currently manage files sizes correctly, so this can be a solution. And it's interesting because iterable based tqdm is faster than the manually updated tqdm.

However, there is an issue: with this approach (iterable based tqdm with blocksize), it will report an incorrect total (incorrect file size) if your file isn't exactly a multiple of blocksize, because the last block isn't full, whereas the `update` approach will always be correct.

So what is bothering me is that tqdm will report a wrong file size. And I cannot think of any way to fix that...

**Comment by casperdcl** [comment]
This will either be fixed in #273 (essentially merging in `rate_scale` logic above into `unit_scale`) or not at all.
